### PR TITLE
Add Parameter Validation

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Parameter validation and update dependencies.

## Commits

- refactor: enhance parameter validation in UlidConverter and Ulid constructors
- chore: update vm2.TestUtilities to version 2.1.0
- chore: update package locks

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: src/Ulid/Serialization.SysJson.Ulid/UlidConverter.cs [hunk 1]
```diff
@@ -24,14 +24,15 @@ public class UlidConverter : JsonConverter<vm2.Ulid>
     /// <param name="_">The <see cref="JsonSerializerOptions"/> to use during serialization. This parameter is not used in this
     /// implementation but is required by the method signature.</param>
     public override void Write(
-        Utf8JsonWriter writer,
+        [NotNull] Utf8JsonWriter writer,
         vm2.Ulid value,
         JsonSerializerOptions _)
     {
+        ArgumentNullException.ThrowIfNull(writer, nameof(writer));
+
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply src/Ulid/Serialization.SysJson.Ulid/UlidConverter.cs]: Removed!

File: src/Ulid/Ulid.cs [hunk 2]
```diff
@@ -116,8 +116,12 @@ public Ulid(ReadOnlySpan<byte> bytes, bool isUtf8)
     /// Initializes a new instance of the <see cref="Ulid"/> struct using the specified string representation.
     /// </summary>
     /// <param name="source">The string representation of the ULID to parse. Must be a valid ULID string.</param>
-    public Ulid(string source)
-        => _ulidBytes = Parse(source)._ulidBytes;
+    public Ulid([NotNull] string source)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(source, nameof(source));
+
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply src/Ulid/Ulid.cs]: restored

File: src/Ulid/Ulid.cs [hunk 3]
```diff
@@ -212,7 +216,12 @@ public Ulid(DateTimeOffset dateTime, ReadOnlySpan<byte> randomBytes)
     /// Defines an explicit conversion from a string to a <see cref="Ulid"/> instance.
     /// </summary>
     /// <param name="s"></param>
-    public static explicit operator Ulid(string s) => new(s);
+    public static explicit operator Ulid([NotNull] string s)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(s, nameof(s));
+
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply src/Ulid/Ulid.cs]: OK
